packages/react-flicking/src/react-flicking/index.tsx

```diff
@@ -77,13 +77,15 @@ export default class Flicking extends React.Component<IFlickingProps> {
     const panel = flicking._conf.panel;
     const horizontal = this.props.horizontal;
     const className = `${this.props.prefix}-panel`;
-    panel.$list = Array.prototype.slice.call(flicking.$container.children);
+    panel.$list = [].slice.call(flicking.$container.children);
     panel.count = panel.$list.length;
-    panel.origCount = panel.count;
+    panel.origCount = [].concat(this.props.children).length;
 
     if (this.props.circular) {
       flicking._arrangePanels();
-      flicking._movePanelPosition(panel.no - panel.index, true);
+      const deleteCount = panel.no - panel.index;
+      const isAppend = deleteCount >= 0;
+      flicking._movePanelPosition(isAppend ? deleteCount : -deleteCount, isAppend);
```

Comment:
What does above code means?

Reply:
@WoodNeck 
panel.no is the index of the panel in the position indicated by Camera.

panel.index is the current panel.

The positions are set in the order of the list, but the alignment is initialized each time you renew.

(panel.no - panel to align.send back or forward as much as index.